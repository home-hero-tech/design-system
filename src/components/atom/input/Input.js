import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import css from './Input.module.scss';

const Input = ({
  id,
  name,
  type,
  placeholder,
  value,
  onChange,
  disabled,
  dark
}) => {
  const isCheckOrRadio = type === 'checkbox' || type === 'radio';

  const classes = classNames({
    [css['c-input']]: true,
    [css['c-input--disabled']]: disabled,
    [css[`c-input__${type}`]]: isCheckOrRadio,
    [css[`c-input__${type}--disabled`]]: disabled && isCheckOrRadio,
    [css['c-input--dark']]: dark
  });

  return (
    <input
      className={classes}
      id={id}
      name={name}
      type={type}
      placeholder={placeholder}
      value={value}
      onChange={onChange}
      disabled={disabled}
    />
  );
};

Input.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string,
  onChange: PropTypes.func,
  placeholder: PropTypes.string,
  type: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  disabled: PropTypes.bool,
  dark: PropTypes.bool
};

Input.defaultProps = {
  id: null,
  name: null,
  onChange: f => f,
  placeholder: null,
  type: 'text',
  value: null,
  disabled: null,
  dark: null
};

export default Input;
