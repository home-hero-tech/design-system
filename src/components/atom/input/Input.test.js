import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Input from './Input';

describe('Input', () => {
  it('should match with the snapshot', () => {
    const wrapper = shallow(<Input placeholder="placeholder text" id="ipt" name="ipt" type="text" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
