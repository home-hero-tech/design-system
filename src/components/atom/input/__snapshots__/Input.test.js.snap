// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match with the snapshot 1`] = `
<div
  className="c-input-wrapper"
>
  <input
    className="c-input"
    disabled={null}
    id="ipt"
    name="ipt"
    onChange={[Function]}
    placeholder="placeholder text"
    type="text"
  />
</div>
`;
